src/earthkit/data/readers/geotiff.py

```diff
@@ -0,0 +1,219 @@
+# (C) Copyright 2024 ECMWF.
+#
+# This software is licensed under the terms of the Apache Licence Version 2.0
+# which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+# In applying this licence, ECMWF does not waive the privileges and immunities
+# granted to it by virtue of its status as an intergovernmental organisation
+# nor does it submit to any jurisdiction.
+#
+
+import mimetypes
+from functools import cached_property
+
+import numpy as np
+
+from ..core.fieldlist import Field
+from ..core.fieldlist import FieldList
+from ..core.geography import Geography
+from ..core.metadata import RawMetadata
+from ..utils.bbox import BoundingBox
+from ..utils.projections import Projection
```

Comment:
Projection should be a local import in the methods that use it. With this we can delay importing e.g. catropy until it is actually used.